vmdb/spec/requests/shared_examples/api.rb

```diff
@@ -0,0 +1,228 @@
+# Methods defining commong expect patterns
+def expect_result_to_include_data(collection, data)
+  expect(@result).to have_key(collection)
+  data.each do |key, value|
+    value_list = fetch_value(value)
+    expect(@result[collection].size).to eq(value_list.size)
+    expect(@result[collection].collect { |r| r[key] }.sort).to eq(value_list.sort)
+  end
+end
+
+def expect_result_to_include_hrefs(collection, hrefs)
+  expect(@result).to have_key(collection)
+  href_list = fetch_value(hrefs)
+  expect(@result[collection].size).to eq(href_list.size)
+  href_list.each do |href|
+    expect(resources_include_suffix?(@result[collection], "href", href)).to be_true
+  end
+end
+
+def expect_result_resources_to_match_key_data(collection, key, values)
+  value_list = fetch_value(values)
+  expect(@result).to have_key(collection)
+  expect(@result[collection].size).to eq(value_list.size)
+  @result[collection].zip(value_list).each do |hash, value|
+    expect(hash).to have_key(key)
+    expect(hash[key]).to match(value)
+  end
+end
+
+def call_custom_expects(method)
+  expect(respond_to?(method)).to be_true
+  public_send(method)
+end
+
+shared_examples_for "bad_request" do |error_message|
+  it "request failed with bad request (400) #{error_message}" do
+    expect(@success).to be_false
+    expect(@code).to eq(400)
+
+    if error_message.present?
+      expect(@result).to have_key("error")
+      expect(@result["error"]["message"]).to match(error_message)
+    end
+  end
+end
+
+shared_examples_for "user_unauthorized" do
+  it "request failed with unauthorized (401)" do
+    expect(@success).to be_false
```

Comment:
1) get rid of @success
2) rest is ok. (although that custom matcher would be cool)